src/NationalInstruments.Analyzers.Utilities/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -104,5 +107,60 @@ public static bool IsOrInheritsFromClass(this ITypeSymbol type, string className
         {
             return type.GetBaseTypesAndThis().OfType<INamedTypeSymbol>().Any(t => t.GetFullName() == className);
         }
+
+        /// <summary>
+        /// Gets all of the public properties of the type.
+        /// </summary>
+        /// <param name="typeSymbol">the symbol to inspect</param>
+        /// <returns>Array of <see cref="IPropertySymbol"/> of the public properties</returns>
+        public static ImmutableArray<IPropertySymbol> GetPublicPropertySymbols(
+            this ITypeSymbol typeSymbol)
+        {
+            return typeSymbol.GetMembers()
+                .Where(m => m.Kind == SymbolKind.Property
+                    && m.DeclaredAccessibility == Accessibility.Public)
+                .Cast<IPropertySymbol>()
+                .ToImmutableArray();
+        }
+
+        /// <summary>
+        /// Gets whether this type is an enumerable.
+        /// </summary>
+        /// <param name="typeSymbol">the symbol to inspect.</param>
+        /// <returns>True if the type implements IEnumerable, false otherwise.</returns>
```

Comment:
Minor: your capitalization of documentation is inconsistent. While I'd prefer SA1628 to be enabled so that we're all consistent, [it's not](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/1057).

Reply:
Capitalization updated